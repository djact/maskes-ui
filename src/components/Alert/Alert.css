.my-alert {
    width: 100%;
    position: absolute !important;
    z-index: 25 !important;
    opacity: 90%;
}