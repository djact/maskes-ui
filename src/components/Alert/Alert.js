import React from 'react';
import { connect } from 'react-redux';
import Alert from 'react-bootstrap/Alert';

const Alert = (props) => {
    const { alerts } = props
    if (alerts !== null && alerts.length > 0) {
        return (
            alerts.map(alert => (
                <Alert key={alert.id} variant={alert.variant}>
                    {alert.msg}
                </Alert>
            ))
        )
    }

};


const mapStateToProps = state => ({
    alerts: state.alert
});

export default connect(mapStateToProps)(Alert);