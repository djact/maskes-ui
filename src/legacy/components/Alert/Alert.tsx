import React from 'react'
import { Alert as AlertBS } from 'react-bootstrap'
import './Alert.css'
interface IAlert {
    id: string
    msg: string
    variant:
        | 'primary'
        | 'secondary'
        | 'success'
        | 'danger'
        | 'warning'
        | 'info'
        | 'dark'
        | 'light'
        | undefined
}
const Alert = ({ alerts }: { alerts: IAlert[] }) => {
    return alerts && alerts.length && alerts.length > 0
        ? alerts.map((alert) => (
              <AlertBS
                  className="my-alert"
                  key={alert.id}
                  variant={alert.variant}
              >
                  {alert.msg}
              </AlertBS>
          ))
        : null
}
export default Alert
