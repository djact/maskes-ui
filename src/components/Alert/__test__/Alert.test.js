import React from 'react';
import { render, shallow } from 'enzyme';
import Alert from '../Alert';

describe('Alert', () => {
	it('should render correctly without props', () => {
		const component = shallow(<Alert />);
		expect(component).toMatchSnapshot();
	});

	it('should render correctly with passed in props', () => {
		const alertData = [
			{ id: '1', msg: 'danger alert', variant: 'danger' },
			{ id: '2', msg: 'success alert', variant: 'success' },
			{ id: '3', msg: 'warning alert', variant: 'warning' }
		];
		const component = render(<Alert alerts={alertData} />);
		expect(component).toMatchSnapshot();
	});
});
