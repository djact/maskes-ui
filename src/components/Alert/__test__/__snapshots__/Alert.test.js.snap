// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert should render correctly with passed in props 1`] = `
Array [
  <div
    class="fade my-alert alert alert-danger show"
    role="alert"
  >
    danger alert
  </div>,
  <div
    class="fade my-alert alert alert-success show"
    role="alert"
  >
    success alert
  </div>,
  <div
    class="fade my-alert alert alert-warning show"
    role="alert"
  >
    warning alert
  </div>,
]
`;

exports[`Alert should render correctly without props 1`] = `""`;
